drivers/gpio/gpio_nrfx.c

```diff
@@ -173,10 +173,12 @@ static int gpio_nrfx_port_set_masked_raw(const struct device *port,
 					 gpio_port_value_t value)
 {
 	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
-	uint32_t value_tmp;
 
-	value_tmp = nrf_gpio_port_out_read(reg) & ~mask;
-	nrf_gpio_port_out_write(reg, value_tmp | (mask & value));
+	const uint32_t set_mask = value & mask;
+	const uint32_t clear_mask = (~set_mask) & mask;
```

Comment:
I think this will be a bit more straightforward:
```suggestion
	const uint32_t clear_mask = (~value) & mask;
```